adapters/coap-vertx-base/src/main/java/org/eclipse/hono/adapter/coap/DeviceRegistryBasedCertificateVerifier.java

```diff
@@ -214,46 +212,42 @@ public void setResultHandler(final HandshakeResultHandler resultHandler) {
     }
 
     @Override
-    public List<CertificateType> getSupportedCertificateType() {
+    public List<CertificateType> getSupportedCertificateTypes() {
         return SUPPORTED_TYPES;
     }
 
     @Override
     public CertificateVerificationResult verifyCertificate(
             final ConnectionId cid,
             final ServerNames serverName,
-            final Boolean clientUsage,
-            final boolean truncateCertificatePath,
-            final CertificateMessage message,
-            final DTLSSession session) {
+            final InetSocketAddress remotePeer,
+            final boolean clientUsage,
+            final boolean verifySubject,
+            final boolean truncateCertificatePath, final CertificateMessage message) {
 
         try {
             final CertPath certChain = message.getCertificateChain();
             if (certChain == null) {
                 final AlertMessage alert = new AlertMessage(AlertLevel.FATAL,
-                        AlertDescription.BAD_CERTIFICATE, session.getPeer());
+                        AlertDescription.BAD_CERTIFICATE);
                 throw new HandshakeException("RPK not supported", alert);
             }
-            final var certificates = certChain.getCertificates();
-            if (certificates.isEmpty()) {
-                final AlertMessage alert = new AlertMessage(
-                        AlertLevel.FATAL,
-                        AlertDescription.BAD_CERTIFICATE,
-                        session.getPeer());
+            final List<? extends Certificate> list = certChain.getCertificates();
+            final int pathSize = list.size();
+            if (pathSize < 1) {
+                final AlertMessage alert = new AlertMessage(AlertLevel.FATAL,
+                        AlertDescription.BAD_CERTIFICATE);
```

Comment:
any particular reason to change this code?

Reply:
In Californium 2.7 the `clientUsage` was `Boolean` and so `if (clientUsage != null) {` is now obsolete. The  other changes may be re-edited towards the previous terms and coding style in hone.

Reply:
yes, please do so

Reply:
the original code style is reverted